pkg/clustermanager/cluster_manager.go

```diff
@@ -527,6 +528,30 @@ func (c *ClusterManager) EKSAClusterSpecChanged(ctx context.Context, cluster *ty
 				return true, nil
 			}
 		}
+	case v1alpha1.CloudStackDatacenterKind:
+		existingCsdc, err := c.clusterClient.GetEksaCloudStackDatacenterConfig(ctx, cc.Spec.DatacenterRef.Name, cluster.KubeconfigFile, newClusterSpec.Cluster.Namespace)
+		if err != nil {
+			return false, err
+		}
+		csDc := datacenterConfig.(*v1alpha1.CloudStackDatacenterConfig)
+		if !existingCsdc.Spec.Equal(&csDc.Spec) {
+			logger.V(3).Info("New provider spec is different from the new spec")
+			return true, nil
+		}
+
+		for _, config := range machineConfigs {
```

Comment:
This loop may not work because it is possible that someone added a new machine config object, so if it looks for it below it will throw an error. I think that is why we had it the way we did before where we check based on the references and not just looping through all the machine configs.

Reply:
Refactored again. Could you take another look?